diff --git sbin/cdb_install_support.sh sbin/cdb_install_support.sh
index e8048df95..f4e734f5d 100755
--- sbin/cdb_install_support.sh
+++ sbin/cdb_install_support.sh
@@ -4,48 +4,99 @@
 # See LICENSE file.
 
 
-MY_DIR=`dirname $0` && cd $MY_DIR && MY_DIR=`pwd`
+declare -g MY_SCRIPT;
+declare -g MY_TOP;
+
+MY_SCRIPT="$(realpath "$0")"
+MY_TOP="${MY_SCRIPT%/*}"
+
+function execute {
+    echo "Executing: $@"
+    eval "$@"
+}
+
+function pushd { builtin pushd "$@" > /dev/null || exit; }
+function popd  { builtin popd  > /dev/null || exit; }
+
+function yes_or_no_to_go
+{
+
+    printf  ">\\n";
+    printf  "> $1\\n";
+    printf  ">\\n";
+    read -p ">> Do you want to continue (y/N)? " answer
+    case ${answer:0:1} in
+	y|Y )
+	    printf ">> We are recreating ...... ";
+	    ;;
+	* )
+            printf ">> Please check your patch in order to backup them. Stop here.\\n";
+	    exit;
+    ;;
+    esac
+
+}
+
+
 if [ -z "${CDB_ROOT_DIR}" ]; then
-    CDB_ROOT_DIR=$MY_DIR/..
+    echo ">> ... CDB_ROOT_DIR is not defined yet."
+    CDB_ROOT_DIR=$(realpath $MY_TOP/..)
+    echo ">> Now CDB_ROOT_DIR is ${CDB_ROOT_DIR}"
+else
+    echo "CDB_ROOT_DIR is already defined as ${CDB_ROOT_DIR}"
 fi
-CDB_INSTALL_DIR=$CDB_ROOT_DIR/..
-CDB_SHORT_HOSTNAME=`hostname -s`
-
-# Check support directory
-if [ -z $CDB_SUPPORT_DIR ]; then
-    export CDB_SUPPORT_DIR=$CDB_INSTALL_DIR/support-$CDB_SHORT_HOSTNAME
-    if [ -d $CDB_SUPPORT_DIR ]; then
-        cd $CDB_SUPPORT_DIR
-	export CDB_SUPPORT_DIR=`pwd`
-    fi
+
+CDB_INSTALL_DIR=$(realpath ${CDB_ROOT_DIR}/..)
+CDB_SHORT_HOSTNAME=$(hostname -s)
+
+if [ -z "${CDB_SUPPORT_DIR}" ]; then
+    echo ">> ... CDB_SUPPORT_DIR is not defined yet."
+    CDB_SUPPORT_DIR="$CDB_INSTALL_DIR/support-$CDB_SHORT_HOSTNAME"
+    echo ">> Now CDB_SUPPORT_DIR is ${CDB_SUPPORT_DIR}"
+else
+    echo "CDB_SUPPORT_DIR is already defined as ${CDB_SUPPORT_DIR}"
 fi
 
-# Check data directory
-if [ -z $CDB_DATA_DIR ]; then
-    export CDB_DATA_DIR=$CDB_INSTALL_DIR/data
-    if [ -d $CDB_DATA_DIR ]; then
-        cd $CDB_DATA_DIR
-	export CDB_DATA_DIR=`pwd`
-    fi
+
+if [ -z "${CDB_DATA_DIR}" ]; then
+    echo ">> ... CDB_DATA_DIR is not defined yet."
+    CDB_DATA_DIR="$CDB_INSTALL_DIR/data"
+    echo ">> Now CDB_DATA_DIR is ${CDB_DATA_DIR}"
+else
+    echo "CDB_DATA_DIR is already defined as ${CDB_DATA_DIR}"
 fi
 
-execute() {
-    echo "Executing: $@"
-    eval "$@"
-}
+
+export CDB_ROOT_DIR;
+export CDB_SUPPORT_DIR;
+export CDB_DATA_DIR;
+export CDB_INSTALL_DIR;
+
 
 echo "Creating new CDB support directory $CDB_SUPPORT_DIR."
-cd `dirname $CDB_SUPPORT_DIR`
-execute cp -R $CDB_ROOT_DIR/support $CDB_SUPPORT_DIR
 
-cd $CDB_SUPPORT_DIR
-echo "Building support in $PWD"
+pushd "${CDB_INSTALL_DIR}"
 
-execute $CDB_SUPPORT_DIR/bin/clean_all.sh
-execute $CDB_SUPPORT_DIR/bin/install_all.sh
+if [ -d "$CDB_SUPPORT_DIR" ]; then
+    yes_or_no_to_go "CDB_SUPPORT_DIR : ${CDB_SUPPORT_DIR} exist. Do you want to recreate it?"
+    rm -rf ${CDB_SUPPORT_DIR}
+fi
+
+printf "%s \\n" "${CDB_SUPPORT_DIR}";
+mkdir -p ${CDB_SUPPORT_DIR}/
+
+pushd "${CDB_SUPPORT_DIR}"
+execute "cp -R ${CDB_ROOT_DIR}/support/bin ."
+popd
+
+# We don't need to clean them, because we just create it from scratch
+# execute "$CDB_SUPPORT_DIR/bin/clean_all.sh"
+execute "$CDB_SUPPORT_DIR/bin/install_all.sh"
 
 if [ ! -d $CDB_DATA_DIR ]; then
     echo "Creating data directories in $CDB_DATA_DIR"
-    mkdir -p "$CDB_DATA_DIR/log"
-    mkdir -p "$CDB_DATA_DIR/propertyValue"
+    mkdir -p "$CDB_DATA_DIR/{log,propertyValue}"
 fi
+
+popd
+
