diff --git tools/developer_tools/cdb_plugins/update_plugin_generated_files.py tools/developer_tools/cdb_plugins/update_plugin_generated_files.py
index 7376782f6..ad0ef2cf6 100755
--- tools/developer_tools/cdb_plugins/update_plugin_generated_files.py
+++ tools/developer_tools/cdb_plugins/update_plugin_generated_files.py
@@ -23,7 +23,7 @@ plugin_manager = PluginManager(cdb_db_name, use_default_storage_directory=True)
 
 update_configuration = False
 if sys.argv.__len__() == 3:
-    input_update_configuration = sys.argv[2]
+    input_update_configuration = int(sys.argv[2])
     if input_update_configuration is not None:
         if input_update_configuration == 1:
             update_configuration = True
diff --git tools/developer_tools/cdb_plugins/utilities/plugin_manager.py tools/developer_tools/cdb_plugins/utilities/plugin_manager.py
index 7ecdc61e4..82efcb9a4 100644
--- tools/developer_tools/cdb_plugins/utilities/plugin_manager.py
+++ tools/developer_tools/cdb_plugins/utilities/plugin_manager.py
@@ -159,6 +159,8 @@ class PluginManager():
             os.mkdir(CDB_JAVA_PLUGIN_PATH)
         if not os.path.exists(CDB_XHTML_PLUGIN_PATH):
             os.makedirs(CDB_XHTML_PLUGIN_PATH)
+        if not os.path.exists(CDB_PYTHON_PLUGIN_PATH):
+            os.makedirs(CDB_PYTHON_PLUGIN_PATH)
         if not os.path.exists(self.cdb_plugin_configuration_storage):
             os.makedirs(self.cdb_plugin_configuration_storage)
 
@@ -191,7 +193,12 @@ class PluginManager():
         if os.path.exists(CDB_PYTHON_PLUGIN_PATH):
             # Create a __init__.py file for all python plugins.
             init_file_path = "%s/%s" % (CDB_PYTHON_PLUGIN_PATH, '__init__.py')
-            init_file = os.open(init_file_path, os.O_CREAT)
+            init_file = os.open(init_file_path, os.O_CREAT|os.O_WRONLY)
+            os.write(init_file, "#!/usr/bin/env python\n")
+            os.write(init_file, "\"\"\"\n")
+            os.write(init_file, "Copyright (c) UChicago Argonne, LLC. All rights reserved.\n")
+            os.write(init_file, "See LICENSE file.\n")
+            os.write(init_file, "\"\"\"")
             os.close(init_file)
 
     @staticmethod
