diff --git db/sql/static/populate_setting_type.sql db/sql/static/populate_setting_type.sql
index 2b78aaa5f..51353c234 100644
--- db/sql/static/populate_setting_type.sql
+++ db/sql/static/populate_setting_type.sql
@@ -493,7 +493,7 @@ description.','true'),
 (25041,'ItemDomainCableInventory.Help.ListPage.Display.Fragment','Display Help guide to the user on the main list page of the Component Instance entity.','true'),
 (25042,'ItemDomainCableInventory.List.Scope.Display','Show list page with a scope applied.','All'),
 (25043,'ItemDomainCableInventory.List.Scope.Display.PropertyTypeId','Setting used along side a display scope of property type.',NULL),
-(25045,'ItemDomainCableInventory.List.Load.FilterDataTable','Automatically load list filter values in data table on page load.','true');
+(25045,'ItemDomainCableInventory.List.Load.FilterDataTable','Automatically load list filter values in data table on page load.','true'),
 (25046,'ItemDomainCableInventory.List.Display.Length','Display column for length in cable inventory list.','true'),
 (25047,'ItemDomainCableInventory.List.FilterBy.Length','Filter for components by length.',NULL);
 /*!40000 ALTER TABLE `setting_type` ENABLE KEYS */;
